core/snippets/snippets/c++.snippet

```diff
@@ -0,0 +1,53 @@
+language: cpp
+---
+
+name: templateDeclaration
+phrase: template
+-
+template <$0>
+---
+
+name: publicSectionDeclaration
```

Comment:
Could you please check the C++ language specification and use names that reflect official terminology? Thanks!